dbms/src/Storages/Transaction/tests/gtest_new_kvstore.cpp

```diff
@@ -0,0 +1,255 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "kvstore_helper.h"
+
+namespace DB
+{
+namespace tests
+{
+TEST_F(RegionKVStoreTest, KVStoreFailRecovery)
```

Comment:
What is the difference between gtest_kvstore and gtest_new_kvstore?

Reply:
gtest_kvstore is old tests of KVStore without a complete mock of proxy.
gtest_new_kvstore is new tests of KVStore which involves pipeline of proxy, like commit/apply etc..